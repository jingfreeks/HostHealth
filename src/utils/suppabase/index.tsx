export {default as supabase} from './suppabase';
