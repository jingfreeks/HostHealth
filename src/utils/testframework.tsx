import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
// import {QueryClient, QueryClientProvider} from '@tanstack/react-query';
// // import {DARK_THEME} from '@/utils/constants';
// // import {ThemeProvider} from 'styled-components/native';

type MockProviderType = {
  children?: React.ReactNode;
  store?: object;
};
export const MockProvider = (props: MockProviderType) => {
  const {store, children} = props;
  const mockStore = configureStore([thunk]);
  const mocked = mockStore(store);
  return <Provider store={mocked}>{children}</Provider>;
};

export const testingProps = (reference: string): unknown => ({
  testID: reference,
  accessibilityLabel: reference,
});

// const queryClient = new QueryClient({
//   defaultOptions: {
//     queries: {
//       retry: false,
//     },
//   },
// });
// export const wrapper = ({children}) => (
//   <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
// );
