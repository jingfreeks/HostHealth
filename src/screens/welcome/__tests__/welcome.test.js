import React from 'react';
import {fireEvent} from '@testing-library/react-native';
import {renderWithProviders} from '@/utils/testframeworknew';
import Welcome from '../welcome';

jest.mock('../hooks', () => {
  return {
    UseWelcomeHooks: () => {
      return {
        handleSignIn: jest.fn(),
        handleSignUp: jest.fn(),
      };
    },
  };
});
describe('Welcome Screen', () => {
  it('Should work as expected to get snapshot', () => {
    const all = renderWithProviders(<Welcome />);
    expect(all.toJSON()).toMatchSnapshot();
  });
  it('Should work to trigger sign in button press', () => {
    const all = renderWithProviders(<Welcome />);
    const el = all.getByTestId('WelcomeButtonGroupTestID');
    fireEvent(el, 'signInPress');
    expect(all.toJSON()).toBeTruthy();
  });

  it('Should work to trigger sign up button press', () => {
    const all = renderWithProviders(<Welcome />);
    const el = all.getByTestId('WelcomeButtonGroupTestID');
    fireEvent(el, 'signUpPress');
    expect(all.toJSON()).toBeTruthy();
  });
});
