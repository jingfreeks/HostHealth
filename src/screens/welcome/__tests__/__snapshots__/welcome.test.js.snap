// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Screen Should work as expected to get snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <RCTSafeAreaView
    style={
      [
        {},
      ]
    }
  >
    <View
      alignType="center"
      style={
        [
          {
            "alignItems": "center",
            "alignSelf": "center",
            "flexDirection": "column",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "marginBottom": 58.85294117647059,
              "marginTop": 58.85294117647059,
            },
          ]
        }
      >
        <RCTSafeAreaView
          style={
            [
              {},
            ]
          }
        >
          <View
            alignType="center"
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "flexDirection": "column",
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "marginBottom": 58.85294117647059,
                    "marginTop": 98.08823529411764,
                  },
                ]
              }
            >
              <Image
                resizeMode="contain"
                source={
                  {
                    "testUri": "../../../src/assets/mainlogo.png",
                  }
                }
                style={
                  [
                    {
                      "height": 196.17647058823528,
                      "width": 451.20588235294116,
                    },
                  ]
                }
              />
            </View>
          </View>
        </RCTSafeAreaView>
        <View
          style={
            [
              {
                "alignItems": "center",
                "marginBottom": 20,
                "marginTop": 20,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#0e3f6c",
                  "fontFamily": "Avenir",
                  "fontSize": 18,
                  "fontWeight": "900",
                },
              ]
            }
          >
            Welcome to Host Healthcare!
          </Text>
          <Text
            style={
              [
                {
                  "color": "#0e3f6c",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "fontWeight": "400",
                },
              ]
            }
          >
            Sign up to find your perfect job today.
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "marginLeft": 5,
            "marginRight": 5,
            "paddingBottom": 5,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
          },
        ]
      }
    >
      <View
        border={10}
        borderc="#99E1E1"
        borderw={2}
        style={
          [
            {
              "borderColor": "#99E1E1",
              "borderRadius": 10,
              "borderWidth": 2,
            },
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#d6f3f3",
              "borderRadius": 10,
              "opacity": 1,
              "paddingBottom": 15,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 15,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#03606A",
                  "fontFamily": "Avenir",
                  "fontSize": 15,
                  "fontWeight": "500",
                },
              ]
            }
          >
            SIGN UP
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "marginLeft": 5,
            "marginRight": 5,
            "paddingBottom": 5,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
          },
        ]
      }
    >
      <View
        border={10}
        borderc="#99E1E1"
        borderw={2}
        style={
          [
            {
              "borderColor": "#99E1E1",
              "borderRadius": 10,
              "borderWidth": 2,
            },
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderRadius": 10,
              "opacity": 1,
              "paddingBottom": 15,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 15,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#03606A",
                  "fontFamily": "Avenir",
                  "fontSize": 15,
                  "fontWeight": "500",
                },
              ]
            }
          >
            SIGN IN
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
